config_ext_test.go

```diff
@@ -455,6 +461,50 @@ func runCommonTests(t *testing.T, provider Provider) {
 		require.NoError(t, provider.Get("occupants.honda").Populate(&c))
 		assert.Equal(t, 1, len(c.Humans.Backseat), "wrong number of backseat occupants")
 	})
+
+	t.Run("non-string scalar keys", func(t *testing.T) {
+		t.Run("numeric key", func(t *testing.T) {
+			var val string
+			run(t, testCase{
+				Value:     provider.Get("1"),
+				HasValue:  true,
+				Interface: "car",
+				Populate:  &val,
+			})
+		})
+
+		t.Run("nil key", func(t *testing.T) {
+			var val string
+			run(t, testCase{
+				Value:     provider.Get("~"),
+				HasValue:  true,
+				Interface: "no car",
+				Populate:  &val,
+			})
+		})
+
+		t.Run("boolean key", func(t *testing.T) {
+			var val string
+			run(t, testCase{
+				Value:     provider.Get("false"),
+				HasValue:  true,
+				Interface: "car unavailable",
+				Populate:  &val,
+			})
+		})
```

Comment:
So...in theory, we should also be able to do `provider.Get("no')`, correct? (I hate this.)

Let's add a test for that.

Reply:
I'll add the test for clarity, because it is a silly case where "no" is tested as a string key instead of being unmarshaled, but it's not fundamentally different than other string key lookups.